fennel/test_lib/mock_client.py

```diff
@@ -150,19 +149,41 @@ def dataset_lookup_impl(
         for col, right_df in data_dict.items():
             right_df[FENNEL_LOOKUP] = True
             right_df[FENNEL_TIMESTAMP] = right_df[timestamp_field]
+            cols_to_replace = []
             # Cast the column in keys to the same dtype as the column in right_df
             for col in keys:
                 if col in right_df and keys[col].dtype != right_df[col].dtype:
                     keys[col] = keys[col].astype(right_df[col].dtype)
+                # Changing dtype of Struct to str for making it hashable
+                if col in right_df and right_df[col].dtype == object:
+                    cols_to_replace.append(col)
+                    right_df[f"{col}__internal"] = right_df[col].apply(
+                        lambda x: str(dict(x))
+                    )
+                    keys[f"{col}__internal"] = keys[col].apply(
+                        lambda x: str(dict(x))
+                    )
+            right_df = right_df.drop(columns=cols_to_replace)
```

Comment:
Is this because window is a struct and also a key ?

Reply:
yup :(, saw this doing in some mock implementation.